code/game/turfs/simulated/floor.dm

```diff
@@ -48,9 +48,11 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
 	return ..()
 
 /turf/simulated/floor/ex_act(severity, target)
-	..()
-	if(is_shielded())
+	if(severity != 1 && is_shielded() && target != src)
+		..()
 		return
+	else
+		..()

```

Comment:
If i don't do it like that, it can break the floor while leaving the protected contents intact. This PR extends the protection the shields give, it's why it's tagged balance.

It is intended that an explosion that breaks an r-window doesn't affect the contents of the floor and the floor itself that are underneath the r-window. is_shielded() can be false after ..() is called, so I can't have that check as it was before, I need to return after calling ..() if the floor is shielded, unless the other conditions (devastating, target == src).